### PR TITLE
Code for checking if the data is outdated.

### PR DESCRIPTION
**Pull request by Vaalmigi**

### CODE REVIEW

File: src/shared/check_outdated_data.c [hunk 1]
```diff
@@ -0,0 +1,29 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <time.h>
+
+int Run(int maxTimeDiff, char *aavild_data) {
```
Comment (frederikja163) [inline src/shared/check_outdated_data.c:5]: This function needs to change. `Run` is **not** a good function name here. Maybe `check_outdated_data` instead?
aavild_data is also not a good name. What is the data and what does it have to do with aavild? aavild is a person not data!

File: src/shared/check_outdated_data.c [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+    time_t cur_time; 
+    //char * aavild_data = "20211122080000";
```
Comment (frederikja163) [inline src/shared/check_outdated_data.c:8]: weird comment. Consider removing this.

File: src/shared/check_outdated_data.c [hunk 3]
```diff
@@ -0,0 +1,29 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <time.h>
+
+int Run(int maxTimeDiff, char *aavild_data) {
+
+    time_t cur_time;
```
Comment (frederikja163) [inline src/shared/check_outdated_data.c:7]: rename this variable to `curTime`

File: src/shared/check_outdated_data.c [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+#include <time.h>
+
+int Run(int maxTimeDiff, char *aavild_data) {
+
+    time_t cur_time; 
+    //char * aavild_data = "20211122080000";
+    struct tm timeptr;
+
+    cur_time = time(NULL);
+    printf("Current timestamp = %ld\n", cur_time);
```
Comment (frederikja163) [inline src/shared/check_outdated_data.c:12]: Why are we printing here?

File: src/shared/check_outdated_data.c [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+    strptime(aavild_data, "%Y%m%d%H%M%S", &timeptr);
```
Comment (frederikja163) [inline src/shared/check_outdated_data.c:14]: please write a comment here describing what this is doing, as the c-code doesn't use nicely named functions it is not clear what you are doing.

File: src/shared/check_outdated_data.c [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    time_t aav_time = mktime(&timeptr);
+    printf("Aavild timestamp = %ld\n", aav_time);
```
Comment (frederikja163) [inline src/shared/check_outdated_data.c:16]: why are we printing here? and stop writing names inside the code like this

File: src/shared/check_outdated_data.c [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+    time_t cur_time; 
+    //char * aavild_data = "20211122080000";
+    struct tm timeptr;
+
+    cur_time = time(NULL);
+    printf("Current timestamp = %ld\n", cur_time);
+
+    strptime(aavild_data, "%Y%m%d%H%M%S", &timeptr);
+    time_t aav_time = mktime(&timeptr);
```
Comment (frederikja163) [inline src/shared/check_outdated_data.c:15]: aav_time is definitely not a good name here. Im not sure at all what this variable is even supposed to be.

File: src/shared/check_outdated_data.c [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+    printf("Aavild timestamp = %ld\n", aav_time);
+
+    double timediff = difftime(cur_time, aav_time);
+    printf("Difference between timestamps = %.0lf\n", timediff);
```
Comment (frederikja163) [inline src/shared/check_outdated_data.c:19]: Why are we printing here?

**Review by frederikja163 — CHANGES_REQUESTED** [review]
It is unclear what your code is doing and all symbols need to be renamed. Generally try not to use names of other coders for variables or functions.

You also need to put comments in the code whenever it isnt clear what your code is doing from just looking at the function names. Since you are working with the c-std lib this happens quite a lot. So rather put too many comments than none at all.

File: src/shared/check_outdated_data.c (hunk 4 above)
Comment (Vaalmigi) [reply src/shared/check_outdated_data.c:12]: just in order for me to see the current timestamp

File: src/shared/check_outdated_data.c (hunk 6 above)
Comment (Vaalmigi) [reply src/shared/check_outdated_data.c:16]: just in order for me to see

Comment (Vaalmigi) [reply src/shared/check_outdated_data.c:16]: but i will just make it a comment

File: src/shared/check_outdated_data.c (hunk 7 above)
Comment (Vaalmigi) [reply src/shared/check_outdated_data.c:15]: I will make a comment explaining this

File: src/shared/check_outdated_data.c (hunk 8 above)
Comment (Vaalmigi) [reply src/shared/check_outdated_data.c:19]: We are printing the difference between the receiving data and the current data. There are no need to print, but it was just for me. I will make the printing part a comment.

File: src/shared/check_outdated_data.c (hunk 1 above)
Comment (Vaalmigi) [reply src/shared/check_outdated_data.c:5]: I will change it sir, please dont hurt me. I will give 2 rupees

File: src/shared/check_outdated_data.c (hunk 2 above)
Comment (Vaalmigi) [reply src/shared/check_outdated_data.c:8]: Yes sir

File: src/shared/check_outdated_data.c (hunk 3 above)
Comment (Vaalmigi) [reply src/shared/check_outdated_data.c:7]: of course sir

File: src/shared/check_outdated_data.c (hunk 5 above)
Comment (Vaalmigi) [reply src/shared/check_outdated_data.c:14]: Will do sir

File: src/shared/check_outdated_data.c [hunk 9]
```diff
@@ -0,0 +1,32 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <time.h>
+
+int check_outdated_data(int maxTimeDiff, char *receivingData) {
+
```
Comment (frederikja163) [inline src/shared/check_outdated_data.c:6]: remove this empty line

File: src/shared/check_outdated_data.c [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+    time_t curTime; 
+
```
Comment (frederikja163) [inline src/shared/check_outdated_data.c:8]: remove this empty line

File: src/shared/check_outdated_data.c [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+    struct tm timeptr;
+
+    curTime = time(NULL);
+   /*  printf("Current timestamp = %ld\n", curTime); */
+
+    /* We are now converting the recieving data(Charachter string) into a timestamp via 
+    this function strptime and this timeptr struct. */
+
```
Comment (frederikja163) [inline src/shared/check_outdated_data.c:16]: remove this empty line

File: src/shared/check_outdated_data.c [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+#include <stdlib.h>
+#include <time.h>
+
+int check_outdated_data(int maxTimeDiff, char *receivingData) {
+
+    time_t curTime; 
+
+    struct tm timeptr;
+
+    curTime = time(NULL);
```
Comment (frederikja163) [inline src/shared/check_outdated_data.c:11]: I would move this to line 7 so it reads
`time_t curTime = time(NULL);`

File: src/shared/check_outdated_data.c [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+   /*  printf("Current timestamp = %ld\n", curTime); */
+
+    /* We are now converting the recieving data(Charachter string) into a timestamp via 
+    this function strptime and this timeptr struct. */
+
+    strptime(receivingData, "%Y%m%d%H%M%S", &timeptr);
+    time_t timetostamp = mktime(&timeptr);
```
Comment (frederikja163) [inline src/shared/check_outdated_data.c:18]: this variable should be called `timeToStamp` instead

File: src/shared/check_outdated_data.c [hunk 14]
```diff
@@ -0,0 +1,32 @@
+#include <stdio.h>
```
Comment (frederikja163) [inline src/shared/check_outdated_data.c:1]: file should be renamed to checkOutdatedData.c like defined in #3

File: src/shared/check_outdated_data.h [hunk 15]
```diff
@@ -0,0 +1 @@
+int check_outdated_data(int maxTimeDiff, char *receivingData);
```
Comment (frederikja163) [inline src/shared/check_outdated_data.h:1]: This file should be called checkOutdatedData.h to comply with #3

You also need a define header pattern like this
```c
#ifndef CHECK_OUTDATED_DATA_H
#define CHECK_OUTDATED_DATA_H

int check_outdated_data(int maxTimeDiff, char *receivingData);

#endif
```

File: src/shared/check_outdated_data.c [hunk 16]
```diff
@@ -0,0 +1,32 @@
...
+#include <stdlib.h>
+#include <time.h>
+
+int check_outdated_data(int maxTimeDiff, char *receivingData) {
```
Comment (frederikja163) [inline src/shared/check_outdated_data.c:5]: This `char*` could be a `const char*` please make it so.

**Review by frederikja163 — CHANGES_REQUESTED** [review]
A few changes still need to be made here. But you are on the right track, it is definitely 10x better now than it was before!

**Review by Ajay2407 — APPROVED** [review]

File: src/shared/check_outdated_data.c (hunk 16 above)
Comment (Vaalmigi) [reply src/shared/check_outdated_data.c:5]: yes

File: src/shared/check_outdated_data.h (hunk 15 above)
Comment (Vaalmigi) [reply src/shared/check_outdated_data.h:1]: yes sir

File: src/shared/check_outdated_data.c (hunk 14 above)
Comment (Vaalmigi) [reply src/shared/check_outdated_data.c:1]: okay sir i will do it sorry for mistake i pay 2 rupees

File: src/shared/check_outdated_data.c (hunk 13 above)
Comment (Vaalmigi) [reply src/shared/check_outdated_data.c:18]: yes will do

File: src/shared/check_outdated_data.c (hunk 12 above)
Comment (Vaalmigi) [reply src/shared/check_outdated_data.c:11]: yessir

File: src/shared/check_outdated_data.c (hunk 11 above)
Comment (Vaalmigi) [reply src/shared/check_outdated_data.c:16]: i anything do sir dont wory 2 rupees i give

File: src/shared/check_outdated_data.c (hunk 10 above)
Comment (Vaalmigi) [reply src/shared/check_outdated_data.c:8]: okay sir

File: src/shared/check_outdated_data.c (hunk 9 above)
Comment (Vaalmigi) [reply src/shared/check_outdated_data.c:6]: yes sir i give u

File: src/shared/checkOutdatedData.c [hunk 17]
```diff
@@ -0,0 +1,27 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <time.h>
+
+int check_outdated_data(int maxTimeDiff, const char *receivingData) {
+    time_t curTime = time(NULL);
+   /*  printf("Current timestamp = %ld\n", curTime); */
+
+    struct tm timeptr;
```
Comment (frederikja163) [inline src/shared/checkOutdatedData.c:9]: should be called timePtr

File: src/shared/checkOutdatedData.c [hunk 18]
```diff
@@ -0,0 +1,27 @@
...
+    /* We are now converting the recieving data(Charachter string) into a timestamp via 
+    this function strptime and this timeptr struct. */
+    strptime(receivingData, "%Y%m%d%H%M%S", &timeptr);
+    time_t timeToStamp = mktime(&timeptr);
+    /* printf("Receving timestamp = %ld\n", timetostamp); */
+
+    double timediff = difftime(curTime, timeToStamp);
```
Comment (frederikja163) [inline src/shared/checkOutdatedData.c:16]: should be called timeDiff

File: src/shared/checkOutdatedData.c (hunk 17 above)
Comment (frederikja163) [reply src/shared/checkOutdatedData.c:9]: Also move this line below the comment.

**Review by frederikja163 — CHANGES_REQUESTED** [review]

**Review by frederikja163 — APPROVED** [review]
Looks good now

**Pull request merged**